netlify/edge-functions/star-search.tsx

```diff
@@ -1,13 +1,27 @@
 import React from "react";
 import { ImageResponse } from "og_edge";
+import { parse, pipe, string, uuid } from "valibot";
 import type { Config } from "https://edge.netlify.com";
 import { getLocalAsset } from "../og-image-utils.ts";
 
 const MAX_CHARS = 130;
+const baseApiUrl = Deno.env.get("NEXT_PUBLIC_API_URL");
+
+const UuidSchema = pipe(string(), uuid());
 
 export default async function handler(req: Request) {
   const { searchParams } = new URL(req.url);
-  const prompt = searchParams.get("prompt");
+  let title: string | undefined = undefined;
+
+  try {
+    const starSearchThreadId = parse(UuidSchema, searchParams.get("id"));
+    const response = await fetch(`${baseApiUrl}/star-search/${starSearchThreadId}`);
+    const thread = await response.json();
+    title = thread?.title;
```

Comment:
Previously, the OG image was the text of the shared prompt. Now it's the title of the thread which is summarized by StarSearch.

If we still want the initial prompt, we can extract it from the thread history, i.e. `thread?.thread_history[0]`.